x-pack/plugins/security/server/routes/authentication/oidc.ts

```diff
@@ -103,183 +77,122 @@ export function defineOIDCRoutes({
-        // Connect Provider sent during implicit authentication flow to the Kibana own proxy page that extracted that URL
-        // fragment and put it into `authenticationResponseURI` query string parameter for this endpoint. See more details
-        // at https://openid.net/specs/openid-connect-core-1_0.html#ImplicitFlowAuth
-        let loginAttempt: ProviderLoginAttempt | undefined;
-        if (request.query.authenticationResponseURI) {
-          loginAttempt = {
-            type: OIDCLogin.LoginWithImplicitFlow,
-            authenticationResponseURI: request.query.authenticationResponseURI,
-          };
-        } else if (request.query.code || request.query.error) {
-          if (isDeprecated) {
-            logger.warn(
-              `The "${serverBasePath}${path}" URL is deprecated and will stop working in the next major version. Use "${serverBasePath}/api/security/oidc/callback" URL instead.`,
-              { tags: ['deprecation'] }
-            );
-          }
-
-          // An HTTP GET request with a query parameter named `code` (or `error`) as the response to a successful (or
-          // failed) authentication from an OpenID Connect Provider during authorization code authentication flow.
-          // See more details at https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth.
-          loginAttempt = {
-            type: OIDCLogin.LoginWithAuthorizationCodeFlow,
-            //  We pass the path only as we can't be sure of the full URL and Elasticsearch doesn't need it anyway.
-            authenticationResponseURI: request.url.pathname + request.url.search,
-          };
-        } else if (request.query.iss) {
-          logger.warn(
-            `The "${serverBasePath}${path}" URL is deprecated and will stop working in the next major version. Use "${serverBasePath}/api/security/oidc/initiate_login" URL for Third-Party Initiated login instead.`,
-            { tags: ['deprecation'] }
-          );
-          // An HTTP GET request with a query parameter named `iss` as part of a 3rd party initiated authentication.
-          // See more details at https://openid.net/specs/openid-connect-core-1_0.html#ThirdPartyInitiatedLogin
-          loginAttempt = {
-            type: OIDCLogin.LoginInitiatedBy3rdParty,
-            iss: request.query.iss,
-            loginHint: request.query.login_hint,
-          };
-        }
+      validate: {
+        query: schema.object(
+          {
+            authenticationResponseURI: schema.maybe(schema.uri()),
+            code: schema.maybe(schema.string()),
+            error: schema.maybe(schema.string()),
+            error_description: schema.maybe(schema.string()),
+            error_uri: schema.maybe(schema.uri()),
+            iss: schema.maybe(schema.uri({ scheme: ['https'] })),
+            login_hint: schema.maybe(schema.string()),
+            target_link_uri: schema.maybe(schema.uri()),
```

Comment:
__note:__ we can remove these as it's only used for [OP initiated login attempts](https://openid.net/specs/openid-connect-core-1_0.html#ThirdPartyInitiatedLogin) and we deprecated usage of `/api/security/oidc/callback` for this since 7.6.0 (https://github.com/elastic/kibana/pull/50695), users should rely on `/api/security/oidc/initiate_login` instead
```suggestion
```